### DISCUSSION TITLE
How to store all form input in a dictionary?

### DISCUSSION THREAD

**Discussion by ledell** [h2oai/wave#1824] [Q&A] [answered]
I am wondering if there's a simple way to store all user input in a dictionary?  The only way I see it do it is to use `name='foo'` and then that input is available `q.app.foo`.  I would love to store all the input in a dictionary, like:  `input_values = {'foo1' : 'val1', 'foo2' : 'val2'}` etc, so it's easier to work with, and I am not sure if there's a direct way to do that.  Thanks!

**Comment by mturoci** [comment, marked as answer]
Hi Erin.

`q.args` is an object of type `Expando`, which is just a [fancy wrapper](https://wave.h2o.ai/docs/arguments/) for python dict. This means

* It's already a dict and might be good enough for your use case as is.
* or it's possible to iterate over `q.args` and store them in your `dict` of choice for further processing.
* or reusing existing wave Expando utils (code below).

```py
from h2o_wave import main, app, Q, ui, copy_expando, Expando, expando_to_dict

# Convert expando to dict.
expando_dict = expando_to_dict(copy_expando(q.args, Expando()))
```

The last approach is considered safest for most of the scenarios since mutating the newly created dict would not change the original Expando (q.args) as well.

**Reply by ledell** [reply]
@mturoci This is great!  The `expando_to_dict` is exactly what I need, thanks.
